Implement Mayhem

nottirb opened the pull request:
- Added a `Dockerfile` for fuzz testing
- Added `Mayhemfile(s)` for fuzz testing
- Added a `mayhem.yml` file for automatically running fuzz tests with Mayhem using the mcode action
- Wrote 2 fuzz harnesses

nottirb committed:
Implement Mayhem

nottirb committed:
Implement Mayhem

nottirb committed:
Fix Crashing Problem

nottirb committed:
Fix Crashing Problem

nottirb committed:
Merge branch 'master' of https://github.com/nottirb/binrw

xansec reviewed (changes requested):
Only request is to remove the image name completely from the Mayhemfiles so they don't get propagated to upstream and accidentally used. Please make the requested changes.

nottirb committed:
Swap out StdRng for Pcg64

nottirb committed:
Remove Image

nottirb commented:
I removed the image name from the Mayhemfiles, and replaced StdRng with Pcg64 to ensure that it generates the same random number on different builds/platforms.

xansec reviewed (approved):
LGTM!
